docs/security.md

```diff
@@ -0,0 +1,187 @@
+# Actions Security Guidelines
+
+> Last updated: **April 24th, 2025**
+
+## Purpose
+
+These guidelines outline security best practices for GitHub Actions workflows and actions used by Hemi Labs.
+
+---
+
+## Key Principles
+
+- **Least privilege** - Always apply the principle of least privilege to workflows, actions and secrets. Limit the scope
+  of permissions and access to only what is strictly necessary.
+
+- **Minimise exposure** - Use minimally scoped secrets and access tokens, reduce the amount of secrets accessible by a
+  workflow, and avoid logging secrets or writing them to files.
+
+- **Integrity and trust** - Use only trusted actions and dependencies, verify their integrity and regularly audit and
+  update them.
+
+---
+
+## 1. Workflows
+
+## 1.1. Secrets
+
+- **Never hardcode secrets** - Always store sensitive data in GitHub Actions secrets and reference them securely within
+  workflows.
+
+- **Limit secret access** - Only expose secrets to the workflows that need them. When possible, avoid using global
+  secrets if specific jobs do not require them. Repository-level secrets or environment secrets can be created to
+  further limit access to the secrets, which can then have a lower scope.
+
+- **Rotate secrets regularly** - Secrets should be rotated periodically to reduce their lifetime and the potential risk
```

Comment:
While I fully agree with this, it feels a little out of scope here I think.  We need a policy for this rather than a best practice (which is inherently a suggestion).

Reply:
Agreed, will remove.